frontend/react/style-guide.md

```diff
@@ -562,34 +578,33 @@ src
   ))}
   ```
 
-  - Always define explicit defaultProps for all undefined props.
-
-  > Why? propTypes are a form of documentation, and providing defaultProps means the reader of your code doesn’t have to assume as much. In addition, it can mean that your code can omit certain type checks.
+  - Always define initial value for all undefined props.
 
-  ```jsx
+  ```tsx
   // bad
-  function SFC({ foo, bar, children }) {
+  interface Props {
+    foo: number;
+    bar: string;
+    children: React.ReactNode;
+  };
+
+  function SFC({ foo, bar, children }: Props) {
     return <div>{foo}{bar}{children}</div>;
   }
-  SFC.propTypes = {
-    foo: PropTypes.number.isRequired,
-    bar: PropTypes.string,
-    children: PropTypes.node,
-  };
+
+  export default SFC;
 
   // good
-  function SFC({ foo, bar, children }) {
+  interface Props {
+    foo: number;
+    bar: string;
+    children: React.ReactNode;
+  };
```

Comment:
```suggestion
  interface Props {
    foo: number;
    bar?: string;
    children?: React.ReactNode;
  };
```

Reply:
I think there may be two good options, 

the first one can be this, put the properties requieres in the interface with default value in the functions, or the second one can be to write `?` in the interface without default values in the function